third_party/libtock-drivers/src/nfc.rs

```diff
@@ -63,34 +65,21 @@ impl NfcTag {
 
     /// Configure the tag type command.
     pub fn configure(tag_type: u8) -> bool {
-        let result_code =
-            syscalls::command(DRIVER_NUMBER, command_nr::CONFIGURE, tag_type as usize, 0);
-        if result_code.is_err() {
-            return false;
-        }
-
-        true
+        syscalls::command(DRIVER_NUMBER, command_nr::CONFIGURE, tag_type as usize, 0).is_ok()
     }
 
-    /// Set the maximum frame delay value to support
-    /// transmission with the reader.
+    /// Set the maximum frame delay value to support transmission with the reader.
     pub fn set_framedelaymax(delay: u32) -> bool {
-        let result_code =
-            syscalls::command(DRIVER_NUMBER, command_nr::FRAMEDELAYMAX, delay as usize, 0);
-        if result_code.is_err() {
-            return false;
-        }
-
-        true
+        syscalls::command(DRIVER_NUMBER, command_nr::FRAMEDELAYMAX, delay as usize, 0).is_ok()
     }
 
     /// 1. Share with the driver a buffer.
     /// 2. Subscribe to having a successful receive callback.
     /// 3. Issue the request for reception.
-    pub fn receive(buf: &mut [u8]) -> bool {
+    pub fn receive(buf: &mut [u8; 256]) -> TransOrRecvStatus {
```

Comment:
You use your own `enum` instead of reusing the libtock result because you want to convert them to something more readable? Why does i.e. `set_framedelaymax` still use `bool` as a return type?

Reply:
> `set_framedelaymax` still use `bool` as a return type?

There's no variations in the kind of errors I would get. It either succeeds and the registers gets set correctly in the driver, or somehow the kernel fails to process the command.

Reply:
But you are using `.is_ok()` instead of `?`. Is the error here not a `TockError`, or are you saying it is not as interesting as the other places where we have `TockError`s?

Reply:
I'm not receiving any information of value from the functions that contain `bool` as return type, so they either succeed or not.

For `receive()` and `transmit()`, I expect something from the kernel, so I want to return that, so I encapsulated the thing I want with the `Result<something, TockError>`. And also to be able to use `?` 😅

Reply:
It's a question of what we want. I'll let @jmichelp and @gendx do a pass of reviews.